### ISSUE TITLE
Add deselect method to igxCalendar

### ISSUE THREAD

**Issue by zdrawku** [IgniteUI/igniteui-angular#2424]
### Description  
Currently we have selectDate method which is selecting date/dates based on the `selection` type. We need to implement similar method for date deselect.

Consider the method to be executed with:
- empty value - this will deselect all selected dates.
- date object - this will deselect the passed date.
- array of date objects - this will deselect all passed dates.

**Comment by tiliev** [comment]
@rkaraivanov @zdrawku 
If I follow the naming pattern of the selection date method, I should name the desired method `deselectDate`. Just calling the method without argument when wanting to deselect all seems a little odd. We do not select everything in the same manner when calling `selectDate`. Should I create a separate method `deselectAll` or should I stick to the instructions in the task and deselect everything when `deselectDate` is called without an argument?

**Comment by zdrawku** [comment]
Agree with the naming. After further dicsussion with @rkaraivanov we've decided only `deselectDate` method to be created, that will cover the scenarios listed in the issue description.

empty value - this will deselect all selected dates.
date object - this will deselect the passed date.
array of date objects - this will deselect all passed dates.

You can go ahead and proceed with the implementation

**Comment by tiliev** [comment]
@rkaraivanov @zdrawku 
Should I add a `deselectDate` method in `igx-date-picker` as well? Since it uses a calendar in single mode the method won't accept any arguments and will just deselect the date. What do you think?

**Comment by zdrawku** [comment]
@tiliev, yes, we should add `deselectDate` method to the `igx-date-picker` as well
